* Don't be stupid insisting on using Apple's Python (for what reason?
  it sucks!)  Don't require a framework, our own python is cute too.
* Gentoo uses .bundle as module extension on OSX.

--- siputils.py
+++ siputils.py
@@ -1362,17 +1362,7 @@
             self.LFLAGS.extend(lflags_plugin)
 
         if sys.platform == "darwin":
-            # We use the -F flag to explictly specify the directory containing
-            # the Python framework rather than rely on the default search path.
-            # This allows Apple's Python to be used even if a later python.org
-            # version is also installed.
-            dl = string.split(sys.exec_prefix, os.sep)
-            try:
-                dl = dl[:dl.index("Python.framework")]
-            except ValueError:
-                error("SIP requires Python to be built as a framework")
-            self.LFLAGS.append("-F%s" % string.join(dl, os.sep))
-            self.LFLAGS.append("-framework Python")
+            self.LFLAGS.append("-lpython%s" % sys.version[:3])
 
         Makefile.finalise(self)
 
@@ -1441,7 +1441,7 @@
             if sys.platform == "win32":
                 ext = "pyd"
             elif sys.platform == "darwin":
-                ext = "so"
+                ext = "bundle"
             elif sys.platform == "cygwin":
                 ext = "dll"
             else:
